tests/regression/runner.py

```diff
@@ -178,20 +186,79 @@ def test_mam_recv_msg(self):
             self.assertTrue(expect_cases[i] in response[i])
 
+    def test_send_transfer(self):
+        # cmd
+        #    0. positive value, tryte maessage, tryte tag, tryte address
+        #    1. zero value, tryte message, tryte tag, tryte address
+        #    2. chinese value, tryte message, tryte tag, tryte address
+        #    3. zero value, chinese message, tryte tag, tryte address
+        #    4. zero value, tryte message, chinese tag, tryte address
+        #    5. negative value, tryte maessage, tryte tag, tryte address
+        #    6. no value, tryte maessage, tryte tag, tryte address
+        #    7. zero value, no maessage, tryte tag, tryte address
+        #    8. zero value, tryte maessage, no tag, tryte address
+        #    9. zero value, tryte maessage, tryte tag, no address
+        #    10. zero value, tryte maessage, tryte tag, unicode address
+        rand_msg = gen_rand_trytes(30)
+        rand_tag = gen_rand_trytes(27)
+        rand_addr = gen_rand_trytes(81)
+        test_cases = [
+            [420, rand_msg, rand_tag, rand_addr],
+            [0, rand_msg, rand_tag, rand_addr],
+            [
+                "生而為人, 我很抱歉".encode(encoding='utf-8'), rand_msg, rand_tag,
+                rand_addr
+            ], [0, "生而為人, 我很抱歉".encode(encoding='utf-8'), rand_tag, rand_addr],
+            [0, rand_msg, "生而為人, 我很抱歉".encode(encoding='utf-8'), rand_addr],
+            [-5, rand_msg, rand_tag], [None, rand_msg, rand_tag, rand_addr],
+            [0, None, rand_tag, rand_addr], [0, rand_msg, None, rand_addr],
+            [0, rand_msg, rand_tag, None], [0, rand_msg, rand_tag, "我思故我在"]
+        ]
+
+        response = []
+        for i in range(len(test_cases)):
+            print("testing case = " + str(test_cases[i]))
+            post_data = "{\"value\":" + str(
+                test_cases[i][0]) + ",\"message\":\"" + str(
+                    test_cases[i][1]) + "\",\"tag\":\"" + str(
+                        test_cases[i][2]) + "\"}"
+            response.append(API("/transaction/", post_data=post_data))
+        # test_case 5 is passed test case since we have only zero transaction in this moment
+        pass_case = [0, 1, 5, 7]
+        for i in range(len(response)):
+            if i in pass_case:
+                res_split = response[i].split(", ")
+                res_json = json.loads(res_split[0])
+
+                # we only send zero tx at this moment
+                self.assertEqual(0, res_json["value"])
+                self.assertEqual(test_cases[i][2], res_json["tag"])
+                self.assertTrue(is_addr_trytes(res_json["address"]))
+                self.assertTrue(
+                    is_addr_trytes(res_json["trunk_transaction_hash"]))
+                self.assertTrue(
+                    is_addr_trytes(res_json["branch_transaction_hash"]))
+                self.assertTrue(is_addr_trytes(res_json["bundle_hash"]))
+                self.assertTrue(is_addr_trytes(res_json["hash"]))
```

Comment:
Assert `signature_and_message_fragment` also?
Remember is has been transfer to trytes.